dnf/transaction_sr.py

```diff
@@ -0,0 +1,600 @@
+                "reason": libdnf.transaction.TransactionItemReasonToString(tsi.reason),
+                "repo_id": tsi.from_repo
+            })
+
+        elif tsi.is_group():
+            group = tsi.get_group()
+
+            group_data = {
+                "action": tsi.action_name,
+                "id": group.getGroupId(),
+                "packages": [],
+                "package_types": libdnf.transaction.compsPackageTypeToString(group.getPackageTypes())
+            }
+
+            for pkg in group.getPackages():
+                group_data["packages"].append({
+                    "name": pkg.getName(),
+                    "installed": pkg.getInstalled(),
+                    "package_type": libdnf.transaction.compsPackageTypeToString(pkg.getPackageType())
+                })
+
+            groups.append(group_data)
+
+        elif tsi.is_environment():
+            env = tsi.get_environment()
+
+            env_data = {
+                "action": tsi.action_name,
+                "id": env.getEnvironmentId(),
+                "groups": [],
+                "package_types": libdnf.transaction.compsPackageTypeToString(env.getPackageTypes())
+            }
+
+            for grp in env.getGroups():
+                env_data["groups"].append({
+                    "id": grp.getGroupId(),
+                    "installed": grp.getInstalled(),
+                    "group_type": libdnf.transaction.compsPackageTypeToString(grp.getGroupType())
+                })
+
+            environments.append(env_data)
+
+    if rpms:
+        data["rpms"] = rpms
+
+    if groups:
+        data["groups"] = groups
+
+    if environments:
+        data["environments"] = environments
+
+    return data
+
+
+class TransactionReplay(object):
+    """
+    A class that encapsulates replaying a transaction. The transaction data are
+    loaded and stored when the class is initialized. The transaction is run by
+    calling the `run()` method, after the transaction is created (but before it is
+    performed), the `post_transaction()` method needs to be called to verify no
+    extra packages were pulled in and also to fix the reasons.
+    """
+
+    def __init__(
+        self,
+        base,
+        fn,
+        ignore_extras=False,
+        ignore_installed=False,
+        skip_unavailable=False
+    ):
+        """
+        :param base: the dnf base
+        :param fn: the filename to load the transaction from
+        :param ignore_extras: whether to ignore extra package pulled into the transaction
+        :param ignore_installed: whether to ignore installed versions of packages
+        :param skip_unavailable: whether to skip transaction packages that aren't available
+        """
+
+        self._base = base
+        self._filename = fn
+        self._ignore_installed = ignore_installed
+        self._ignore_extras = ignore_extras
+        self._skip_unavailable = skip_unavailable
+
+        if self._base.conf.skip_broken:
```

Comment:
Please could use a value from self._base.conf.strict. It is directly related to skip_broken and it is more convenient way in dnf. Skip_broken option is only compat option with yum that is use only to set strict option in dnf.

Reply:
Sorry, must have missed this one when changing this. Though I've got to say the `skip_broken` is more descriptive than `strict`...

Fixed.